File: 백준/Silver/1244. 스위치 켜고 끄기/스위치 켜고 끄기.py
N = int(input())
switch = list(map(int, input().split()))
M = int(input()) # 학생 수
for _ in range(M):
    s, n = map(int, input().split())

    if s == 1:
        for i in range(n-1, N, n):
            if switch[i]:
                switch[i] = 0
            else:
                switch[i] = 1
    else:
        n -= 1
        left = right = n
        while left >= 0 and right < N:
            if switch[left] == switch[right]:
                if switch[left]:
                    switch[left] = 0
                    switch[right] = 0
                else:
                    switch[left] = 1
                    switch[right] = 1
            else:
                break
            left -= 1
            right += 1

if N <= 20:
    print(*switch)
else:
    for i in range(len(switch)):
        print(switch[i], end=" ")
        if i+1 >= 20 and (i+1)%20 == 0:
            print()

File: 백준/Silver/1244. 스위치 켜고 끄기/스위치 켜고 끄기.js
const input = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n");

const number = Number(input.shift());
let arr = input.shift().split(" ").map(Number);

const changeNumber = (num) => (num === 1 ? 0 : 1);

input.forEach((el) => {
  const [gender, switchNumber] = el.split(" ").map(Number);

  if (gender === 1) {
    for (let i = switchNumber; i <= number; i += switchNumber) {
      arr[i - 1] = changeNumber(arr[i - 1]);
    }
  } else {
    let num = 0;
    while (
      switchNumber - num - 1 >= 0 &&
      switchNumber + num - 1 < number &&
      arr[switchNumber - num - 1] === arr[switchNumber + num - 1]
    ) {
      arr[switchNumber - num - 1] = changeNumber(arr[switchNumber - num - 1]);
      arr[switchNumber + num - 1] = changeNumber(arr[switchNumber + num - 1]);
      num++;
    }
    arr[switchNumber - 1] = changeNumber(arr[switchNumber - 1]);
  }
});

for (let i = 0; i < arr.length; i += 20) {
  console.log(arr.slice(i, i + 20).join(" "));
}
